Added TSH Web based controls

ZombieHDGaming opened the pull request:
_**NOTE:**_ This commit looks really large, it honestly isn't, I just didn't feel like dealing with the hassle of Git with other changes I made, so I just uploaded the file via GitHub to get this up here for review.

**FEATURES:**
- _NEW_ Web based controls to control TSH outside of the tool!
- Allows stream runners/users of the tool make Stream Deck/Arduino buttons to send a GET request (or open the page in the background) Ex. http://192.168.1.2:5000/team[NUMBER]-score<UP/DOWN> (You can specify team number and whether you want to make the score go up or down by 1. Going down seems to be stop at 1 but not 0, will need to investigate it later.)
- Allows Resetting and Clearing the score widget (Reset = Scores to 0, Clear = Reset with match and phase cleared)
   - Ex. http://192.168.1.2:5000/reset-<scores/match> (Scores does only as the name suggests, match does whole score widget)
- A new dynamic endpoint was added to allow setting in more advanced information in the score widget!
   - Best Of Amount Ex. http://192.168.1.2:5000/set?best-of=10
   - Phase Name Ex. http://192.168.1.2:5000/set?phase=Top 128 (Yes, spaces work in the names)
   - Match Name Ex. http://192.168.1.2:5000/set?match=Winners Semis (Also yes, spaces work in the names)
   - And they can be combined for setting it all at once for a monster update! Ex. http://192.168.1.2:5000/set?best-of=5&phase=Top 32&match=Winners Finals

This is just for basic functionality, as suggested in issue #79, this is meant only as a stepping stone to open up the possibility for a REST API to maybe make the tool controllable via a web page or the like.

joaorb64 commented:
Woah! This is so nice!
I'm working on backporting the improvements I did for Ultra Hard Tournament before I review the thumbnail integration and then I'll come and review this PR, so it might take a few days . But this looks super hype, can't wait to test it!

joaorb64 commented:
Upon review, I believe it would be better if we could somehow transfer the web server to a separate file, since it now becomes something more general to the whole program!

ZombieHDGaming commented:
When I get home later today, I will see what all would be required for this to be moved to it's separate file. As I don't believe the Stage widget interfaces with the Web server, we may be able to just abstract it outright, but this is just at a first glance, I may be wrong, but definitely a good and feasible idea!

joaorb64 commented:
We'll definitely have to pass some references back and forth, unfortunately :P

ZombieHDGaming commented:
Alright, made the proper changes and moved the web server to its own file. It appears to be working fine after shifting it, just cleaned up the Stage widget as it also had unused imports and shifted the WebServer class, please feel free to test the changes and feel free to re-review the PR!

joaorb64 commented:
Seems great to me! Merging~
Thanks a lot for the contribution!
